haystack/preview/components/samplers/top_p.py

```diff
@@ -89,7 +91,7 @@ def run(self, documents: List[Document], top_p: Optional[float] = None):
         cumulative_probs = torch.cumsum(sorted_probs, dim=-1)
 
         # Find the indices with cumulative probabilities that exceed top_p
-        top_p_indices = torch.where(cumulative_probs <= top_p)[0]
+        top_p_indices = torch.where(torch.BoolTensor(cumulative_probs <= (top_p + epsilon)))[0]
```

Comment:
The need for an epislon is very common for float comparisons. I'd recommend using the native `torch.isclose` function (docs [here](https://pytorch.org/docs/stable/generated/torch.isclose.html)) instead of doing a custom implementation.

Reply:
However, I'm not sure how I would use `torch.isclose` elegantly here without needing to break the `<=` into two separate checks, so I think the epislon here is fine to keep.